storage/gcsemu/util.go

```diff
@@ -59,3 +61,68 @@ func mustJson(val interface{}) []byte {
 	}
 	return b
 }
+
+// RequestHost returns the host from an http.Request, respecting proxy headers. Works locally with devproxy
+// and gulp proxies as well as in AppEngine (both real GAE and the dev_appserver).
+func requestHost(req *http.Request) string {
+	// proxies like gulp are supposed to accumulate original host, next-step-host, etc in order from
+	// client-most to server-most in X-ForwardedHost; return the first entry from that if any are listed
+	if proxyHost := req.Header.Get("X-Forwarded-Host"); proxyHost != "" {
+		// Use the first (closest to client) host
+		splits := strings.SplitN(proxyHost, ",", 2)
+		return splits[0]
+	}
+
+	// Forwarded is the standardized version of X-Forwarded-Host.
+	f := parseForwardedHeader(req.Header.Get("Forwarded"))
+	if len(f.Host) > 0 && len(f.Host[0]) > 0 {
+		return f.Host[0]
+	}
+
+	// Clients that generate HTTP/2 requests should use the :authority header instead
+	// of Host. See http://httpwg.org/specs/rfc7540.html#rfc.section.8.1.2.3
+	host := req.Header.Get("Authority")
+	if len(host) > 0 {
+		return host
+	}
+
+	// Fall back to the host line.
+	return req.Host
+}
+
+// forwarded represents the values of a Forwarded HTTP header.
+//
+// For more details, see the RFC: https://tools.ietf.org/html/rfc7239 and
+// MDN: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded.
+type forwarded struct {
+	By    []string
+	For   []string
+	Host  []string
+	Proto []string
+}
+
+var (
+	forwardedHostRx = regexp.MustCompile(`(?i)host=(.*?)(?:[,;\s]|$)`)
+)
+
+func removeDoubleQuotes(s string) string {
+	return strings.TrimSuffix(strings.TrimPrefix(s, `"`), `"`)
+}
+
+// TODO(nishanth): This currently only supports the forwarded.Host field.
```

Comment:
Is this todo correct? Just validating if it wasnt a copy/paste stuff

Reply:
It was copy paste; the TODO is valid, but I shouldn't assign it to @nishanths